src/tequila/simulators/simulator_pyquil.py

```diff
@@ -236,7 +289,24 @@ def __init__(self, abstract_circuit: QCircuit, variables, use_mapping=True, nois
                 'depolarizing': depolarizing_map
             }
 
-            self.circuit = self.build_noisy_circuit(self.circuit, self.noise)
+            if isinstance(self.noise,str):
+                if device is not None:
+                    if self.noise == 'device':
```

Comment:
simplify block:

if device is not None:
    if not self.noise in ['device', None]:
        raise TequilaPyquilException("Device {} demanded, noise can be None or 'device' but was {}".format(device, self.noise))
    self.device=retrieve_pyquil_device(device,self.noise == "device")

Reply:
There's no reason why an emulated device HAS to use the device's own noise. We can't ban user-made noise  for devices because in cirq we can't emulate device noise, only gate set and connectivity, so users will have to supply their own noise there.